### ISSUE TITLE
GPIO `C-HIDDEN` violations

### ISSUE THREAD

**Issue by MabezDev** [esp-rs/esp-hal#2697]
- Things to make private or `doc-hidden`
    - AlternateFunction
    - Pin trait (anything but number,mask,degrade cannot be called by the user - everything but number + degrade is doc hidden)
    - InputPin trait (all functions are doc-hidden)
    - OutputPin trait (all functions are doc-hidden)
    - AnalogPin (functions are doc-hidden)
    - TouchPin (functions are doc-hidden)
    - GpioRegisterAccess? (doc-hidden, functions are user-callable but should they?)

**Comment by bjoernQ** [comment]
I'm not sure about hiding the traits anymore - they might be useful to users as trait bounds?

**Comment by bjoernQ** [comment]
we still need to agree if we want the mentioned traits to be hidden or not - if not #2856 should be enough to close this

worst case we can just hide them now and see if someone complains - we can make anything public at any point of time later but not the other way around

**Comment by bugadani** [comment]
I don't think we want any of these hidden - although InputPin and OutputPin are implemented for pretty much all pins except on ESP32, users might need to refer to them still if they create pin drivers hidden behind some abstraction.

`GpioRegisterAccess` is hidden right now. Other HALs expose the port of the pins, but that isn't terribly useful for us publicly at the moment. We can mark it as unstable I think, but I'm not sure if it needs to be public at all at least initially.

The rest are unstable so I don't care at the moment.

**Comment by MabezDev** [comment]
I believe this is implemented in https://github.com/esp-rs/esp-hal/pull/2926
